src/NumericalAlgorithms/Spectral/Spectral.hpp

```diff
@@ -419,6 +419,47 @@ template <typename T>
 Matrix interpolation_matrix(const Mesh<1>& mesh,
                             const T& target_points) noexcept;
 
+// @{
+/*!
+ * \brief Matrices that interpolate to the lower and upper boundaries of the
+ * element.
+ *
+ * Assumes that the logical coordinates are \f$[-1, 1]\f$. The first element of
+ * the pair interpolates to \f$\xi=-1\f$ and the second to \f$\xi=1\f$. These
+ * are just the Lagrange interpolating polynomials evaluated at \f$\xi=\pm1\f$.
+ * For Gauss-Lobatto points the only non-zero element is at the boundaries
+ * and is one and so is not implemented.
+ *
+ * \warning This can only be called with Gauss points.
+ */
+const std::pair<Matrix, Matrix>& boundary_interpolation_matrix(
```

Comment:
I find it confusing that this function returns two matrices but has a singular name. I suggest renaming to `boundary_interpolation_matrices`